packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py

```diff
@@ -41,47 +63,53 @@ async def push(file_or_folder: Path, rename_to: Optional[str] = None):
             compress=False,  # disabling compression for faster speeds
             store_relative_path=True,
         )
-        return await _push_file(archive_file_path, None)
+        return await _push_file(user_id, project_id, node_uuid, archive_file_path, None)
 
 
-async def _pull_file(file_path: Path):
-    s3_object = _create_s3_object(file_path)
+async def _pull_file(user_id: int, project_id: str, node_uuid: str, file_path: Path):
+    s3_object = _create_s3_object(project_id, node_uuid, file_path)
     log.info("pulling data from %s to %s...", s3_object, file_path)
     downloaded_file = await filemanager.download_file_from_s3(
-        store_id=0, s3_object=s3_object, local_folder=file_path.parent
+        user_id=user_id,
+        store_id="0",
+        s3_object=s3_object,
+        local_folder=file_path.parent,
     )
     if downloaded_file != file_path:
         if file_path.exists():
             file_path.unlink()
-        move(downloaded_file, file_path)
+        move(f"{downloaded_file}", file_path)

```

Comment:
just out of curiousity is there a difference between `f"{var}"` and str(var)

Reply:
you mean performance wise? I'm not sure. but I like the f strings better;)